### PR TITLE
misc: nil checks and remove deprecated default values

### PR DESCRIPTION
**Pull request by ellemouton**
1) The `Node1Policy` and `Node2Policy` members of the `lnrpc.ChannelEdge` struct can be nil 
and so we should check for this before dereferencing.  ran into panics [here](https://github.com/lightningnetwork/lnd/actions/runs/11898396588/job/33154809901)

2) remove default values for some deprecated log config options

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -562,8 +562,6 @@ func DefaultConfig() Config {
 		LetsEncryptDir:    defaultLetsEncryptDir,
 		LetsEncryptListen: defaultLetsEncryptListen,
 		LogDir:            defaultLogDir,
-		MaxLogFiles:       build.DefaultMaxLogFiles,
```
[the remark this answers is not in the document]
Comment (ellemouton) [reply config.go:565]: works for me. Make sure you definitely are not setting these somewhere else (command line or conf file) and that you definitely have checked out the correct commit

**Review by bhandras — APPROVED** [review]
LGTM 🍫

**Review by yyforyongyu — APPROVED** [review]
LGTM🦾

File: config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Abdulkbk) [reply config.go:565]: I noticed I was setting them in `lnd.conf` 👍

File: lntest/harness_assertion.go [hunk 2]
```diff
@@ -255,10 +255,10 @@ func (h *HarnessTest) AssertNumActiveEdges(hn *node.HarnessNode,
 	// filterDisabled is a helper closure that filters out disabled
 	// channels.
 	filterDisabled := func(edge *lnrpc.ChannelEdge) bool {
-		if edge.Node1Policy.Disabled {
+		if edge.Node1Policy != nil && edge.Node1Policy.Disabled {
```
Comment (yyforyongyu) [inline lntest/harness_assertion.go:258]: hmm looking at this again, why would we ever have an `edge` that has no policy? Did you run into #7261 in the itest?

Comment (ellemouton) [reply lntest/harness_assertion.go:258]: > why would we ever have an edge that has no policy?

we get ChannelUpdates after ChannelAnnouncements. So we may have the announcement and not yet have any updates for it

Comment (ellemouton) [reply lntest/harness_assertion.go:258]: > Did you run into https://github.com/lightningnetwork/lnd/issues/7261 in the itest?

dont think so no. See the panics in the linked build

Comment (ellemouton) [reply lntest/harness_assertion.go:258]: also see other usages of these values  - we always nil-check them

**Pull request merged**